helper/funcs_test.go

```diff
@@ -157,3 +159,47 @@ func BenchmarkCleanEnvVar(b *testing.B) {
 		CleanEnvVar(in, replacement)
 	}
 }
+
+func TestCheckNamespaceScope(t *testing.T) {
+	cases := []struct {
+		desc      string
+		provided  string
+		requested []string
+		offending []string
+	}{
+		{
+			desc:      "root ns requesting namespace",
+			provided:  "",
+			requested: []string{"engineering"},
+		},
+		{
+			desc:      "matching parent ns with child",
+			provided:  "engineering",
+			requested: []string{"engineering", "engineering/sub-team"},
+		},
+		{
+			desc:      "mismatch ns",
+			provided:  "engineering",
+			requested: []string{"finance", "engineering/sub-team", "eng"},
+			offending: []string{"finance", "eng"},
+		},
+		{
+			desc:      "mismatch child",
+			provided:  "engineering/sub-team",
+			requested: []string{"engineering/new-team"},
```

Comment:
Add matching cases for when `provided` is a sub-path:

```suggestion
			requested: []string{"engineering/new-team", "engineering/sub-team", "engineering/sub-team/child"},
```